src/lib/markbind/src/parser.js

```diff
@@ -197,193 +178,19 @@ class Parser {
-              + `\nMissing reference in ${element.attribs[ATTRIB_CWF]}`);
-            this._onError(error);
-            return Parser.createErrorNode(element, error);
-          }
-        }
-        if (isOptional) {
-          // optional includes of segments have now been handled, so delete the attribute
-          delete element.attribs.optional;
-        }
-        if (isIncludeSrcMd) {
-          if (context.mode === 'include') {
-            actualContent = isInline ? actualContent : utils.wrapContent(actualContent, '\n\n', '\n');
-          } else {
-            actualContent = md.render(actualContent);
-          }
-          actualContent = Parser._rebaseReferenceForStaticIncludes(actualContent, element, config);
-        }
-        const wrapperType = isInline ? 'span' : 'div';
-        element.children
-          = cheerio.parseHTML(
-            `<${wrapperType} data-included-from="${filePath}">${actualContent}</${wrapperType}>`,
-            true);
-      } else {
-        let actualContent = (fileContent && isTrim) ? fileContent.trim() : fileContent;
-        if (isIncludeSrcMd) {
-          if (context.mode === 'include') {
-            actualContent = isInline ? actualContent : utils.wrapContent(actualContent, '\n\n', '\n');
-          } else {
-            actualContent = md.render(actualContent);
-          }
-        }
-        const wrapperType = isInline ? 'span' : 'div';
-        element.children
-          = cheerio.parseHTML(
-            `<${wrapperType} data-included-from="${filePath}">${actualContent}</${wrapperType}>`,
-            true);
-      }
-      if (element.children && element.children.length > 0) {
-        if (childContext.callStack.length > CyclicReferenceError.MAX_RECURSIVE_DEPTH) {
-          const error = new CyclicReferenceError(childContext.callStack);
-          this._onError(error);
-          return Parser.createErrorNode(element, error);
-        }
-        element.children = element.children.map(e => self._preprocess(e, childContext, config));
-      }
-    } else if ((element.name === 'panel') && hasSrc) {
-      if (!isUrl && includeSrc.hash) {
-        element.attribs.fragment = includeSrc.hash.substring(1); // save hash to fragment attribute
-      }
-      element.attribs.src = filePath;
-      this.dynamicIncludeSrc.push({ from: context.cwf, to: actualFilePath, asIfTo: filePath });
-      return element;
-    } else if (element.name === 'variable' || element.name === 'import') {
-      return Parser.createEmptyNode();
-    } else {
-      if (element.name === 'body') {
-        // eslint-disable-next-line no-console
-        console.warn(`<body> tag found in ${element.attribs[ATTRIB_CWF]}. This may cause formatting errors.`);
-      }
-      if (element.children && element.children.length > 0) {
-        element.children = element.children.map(e => self._preprocess(e, context, config));
-      }
+  extractInnerVariablesIfNotProcessed(content, childContext, config, filePathToExtract) {
```

Comment:
To allow `componentPreprocessor` and `_extractInnerVariables` to share this common code